### PR TITLE
Add ability to run python modules

### PR DESCRIPTION
**Pull request by jtroe**
Currently please only executes python as standalone scripts.  If I want to import other parts of my library as-is, I have to add the other section to my PYTHONPATH.  Not ideal.

Enter running as modules (with the `-m` flag).  Set the configuration value `PythonMode` in your `please.exe.config` and magically `__init__.py` files are ignored and paths are converted to dot notation.  And I can import relative!

### CODE REVIEW

**Comment by gregoryjscott** [comment]
Please excuse my Python ignorance - can you give me an example file and directory structure of your use case of using modules? Let's say there is a directory named `migrations` where you would normally stick standalone Python scripts. How would it look different if you're using modules? I'd want to build some tests using real files. The way tests are set up in this project is kinda crazy so I'd be happy to help with that.

Is it absolutely necessary to have the extra option for modules? I'm not sure this is a good idea, but would it be possible to just examine what's in the `migrations` directory and do the right thing? Or maybe we just assume modules are in play all the time if that's the typical use case? Is there a need to support standalone scripts AND modules?

Assuming the modules option is necessary then I'd rather implement as a "plain English" option in the same spirit as all the other `please` options rather than the `-m` option. Something like `please run py using modules in directory .\migrations`. (The original idea of "please" was to replace the cryptic command line call/options typically found in CI scripts with more English looking commands that are not only more readable but also polite, hence the name.) Again, configuring new options in the please codebase isn't exactly straight-forward but I can help there too.

In summary, I'm all about making this work better for Python scenarios but you might have to do a little Python mentoring to make it happen.

**Comment by jtroe** [comment]
Python with Modules

```
.
├── __init__.py
├── migrations
|   ├── __init__.py
|   └── 20151004224000_my_super_cool_script_to_run_as_a_module.py
├── lib
|   ├── __init__.py
|   └── my_task_runner_module.py
```

Python, no modules

```
.
├── migrations
|   └── 20151004224000_my_super_cool_script.py
```

The problem I am currently trying to solve is that I would like to be able to import the `my_task_runner_module` module in `my_super_cool_script`, but the only (decent) way (I'm very familiar with) to do that without the module structure is to modify the `PYTHONPATH` environment variable to include the path of `lib`.

**Comment by jtroe** [comment]
It is theoretically possible to examine the directory to see which action to take.  Basically if there is an `__init__.py` file at every level of the directory structure from where you're beginning execution to the desired script, it's a module (or can be executed as such).

Not sure if you recall, but in the last project we worked on together, we ran all the pythons as modules so that way we could have a shared codebase within the project for repeated tasks.

Here's a more complex directory structure:

```
.
├── __init__.py
├── Scripts
|   ├── __init__.py
|   ├── lib
|   |   ├── __init__.py
|   |   └── my_task_runner_module.py
|   └── py
|   |    ├── __init__.py
|    |   └── migrations
|    |    |   ├── 20151004224000_my_super_cool_script_to_run_as_a_module.py
|    |    |   └── __init__.py
```

If `__init__.py` were missing from `.`, `./Scripts`, `./Scripts/py`, or `./Scripts/py/migrations`, and you ran `please.exe` from the context of `/`, then running `python -m Scripts.py.migrations.20151004224000_my_super_cool_script_to_run_as_a_module` would fail.

Also, if in `20151004224000_my_super_cool_script_to_run_as_a_module.py` there was an `import Scripts.lib.my_task_runner_module` statement, and you ran it not as a module, the import would fail, unless there was also a change to the `PYTHONPATH` to include `Scripts/lib`.

**Comment by jtroe** [comment]
If your `please.exe.config` is in the scope of your project (as it has been for all instances I've personally used please), then I think it's reasonable to have a setting to determine whether we're running as script or module.

With that being said, I don't think there is an instance in which I would use please for python in which I would not run as modules.  For my purposes I need to reuse code often.

**Comment by gregoryjscott** [comment]
Thanks for the explanation. I played arounds with python on the command line and I think I get it now.

I agree the setting in `please.exe.config` is a reasonable solution to setting the python mode. I want to walk away from the XML config file (Microsoft is doing the same in vNext) and instead pass the configuration to `please` as JSON but that's another topic. This is the way for now.

File: app/Library/Scripts/Tasks/RunScripts.cs [hunk 1]
```diff
@@ -42,5 +42,12 @@ public override void Execute()
                 }
             }
         }
+
+        private string GetPythonArguments(string fileName)
+        {
+            if (Config.PythonMode == PythonMode.Script) return fileName;
+            var extension = Path.GetExtension(fileName);
+            return String.Format("-m {0}", fileName.Replace("\\", ".").Substring(0, fileName.Length - extension.Length));
+        }
```
Comment (gregoryjscott) [inline app/Library/Scripts/Tasks/RunScripts.cs:51]: I'm pretty sure this function will return the wrong filename on non-Windows because of the `.Replace("\\", ".")`. I think the best way to fix this is to factor GetPythonArguments to be a Task (sub-task of RunScripts) and add some tests to make sure that it returns the correct arguments string given a python mode and filename. The logic will need to be changed to use the `System.IO.Path` class to deal with the filename in a cross-platform way.

This is really the only missing set of tests. I'd like to have some full integration tests that use actual files but it's really not necessary to test that python works. We just need to make sure `please` knows how to call python correctly which `GetPythonArgumentsTest` would do.
